tools/tools_test.go

```diff
@@ -1,673 +1,619 @@
-			output:           `<tool_call> [{"name": "get_current_weather", "parameters": {"format":"fahrenheit","location":"San Francisco, CA"}} {"name": "get_current_weather", "parameters": {"format":"celsius","location":"Toronto, Canada"}}]</tool_call>`,
-			expectedToolCall: []api.ToolCall{t1, t2},
-			expectedTokens:   `<tool_call>`,
+			name:  "whitespace only",
+			input: "   \n\t  ",
+			want:  nil,
 		},
 		{
-			name:             "model without prefix in template, prefix in output, individual tool calls",
-			model:            "llama3.2",
-			output:           `<tool_call> {"name": "get_current_weather", "parameters": {"format":"fahrenheit","location":"San Francisco, CA"}} {"name": "get_current_weather", "parameters": {"format":"celsius","location":"Toronto, Canada"}}`,
-			expectedToolCall: []api.ToolCall{t1, t2},
-			expectedTokens:   `<tool_call>`,
+			name:  "unbalanced braces - missing closing",
+			input: `{"format": "fahrenheit", "location": "San Francisco"`,
+			want:  nil,
 		},
 		{
-			name:             "model with prefix in template, no prefix in output, tokens before",
-			model:            "qwen2.5",
-			output:           `some tokens before [{"name": "get_current_weather", "arguments": {"format":"fahrenheit","location":"San Francisco, CA"}} {"name": "get_current_weather", "arguments": {"format":"celsius","location":"Toronto, Canada"}}]`,
-			expectedToolCall: []api.ToolCall{},
-			expectedTokens:   `some tokens before [{"name": "get_current_weather", "arguments": {"format":"fahrenheit","location":"San Francisco, CA"}} {"name": "get_current_weather", "arguments": {"format":"celsius","location":"Toronto, Canada"}}]`,
+			name:  "unbalanced braces - extra closing",
+			input: `{"format": "fahrenheit"}}`,
+			want: map[string]any{
+				"format": "fahrenheit",
+			},
 		},
 		{
-			name:             "model with prefix in template, prefix in output, tokens after",
-			model:            "qwen2.5",
-			output:           `<tool_call>[{"name": "get_current_weather", "arguments": {"format":"fahrenheit","location":"San Francisco, CA"}} {"name": "get_current_weather", "arguments": {"format":"celsius","location":"Toronto, Canada"}}]</tool_call> some tokens after`,
-			expectedToolCall: []api.ToolCall{t1, t2},
-			expectedTokens:   "",
+			name:  "invalid JSON",
+			input: `{format: fahrenheit, location: "San Francisco"}`,
+			want:  nil,
 		},
 		{
-			name:             "model without prefix in template, no prefix in output, tokens after",
-			model:            "llama3.2",
-			output:           `[{"name": "get_current_weather", "parameters": {"format":"fahrenheit","location":"San Francisco, CA"}} {"name": "get_current_weather", "parameters": {"format":"celsius","location":"Toronto, Canada"}}]</tool_call> some tokens after`,
-			expectedToolCall: []api.ToolCall{t1, t2},
-			expectedTokens:   "",
+			name:  "valid arguments field",
+			input: `{"name": "get_temperature", "arguments": {"format": "fahrenheit", "location": "San Francisco, CA"}}`,
```

Comment:
from above ^ - what do multiple tool calls look like in this case? where there name and args in a list